### ISSUE TITLE
Revisit markup for Bento / SearchWorks search "All / Catalog / Articles" dropdown from scratch

### ISSUE THREAD

**Issue by camillevilla** [sul-dlss/SearchWorks#2516]
This is going to be an iterative process and require some back and forth with JV.

![Screen Shot 2020-08-07 at 2 14 50 PM](https://user-images.githubusercontent.com/5402927/89689048-60553380-d8b8-11ea-9d0f-57c407c9ea8f.png)

### Concerns:
- `<button>` nested inside of SearchWorks `<h1>` does weird things to screenreader nav. Let's move the button out of the `<h1>`. Do we have better options for our HTML now that we have a better command of flexbox?
- remove sr-only print heading to clean up navigation tree
```
<div class="visible-print print-header pull-right">
  <h1>SearchWorks Catalog</h1>
  <h2>Stanford Libraries</h2>
</div>
```

Because this pattern is used in both SearchWorks, we ideally want to make this change in both applications this workcycle.

**Comment by jkeck** [comment]
A few questions from looking into this.

* Should the content that's in the h1 now still remain (e.g. `search catalog` and `search articles+`) as an h1 heading, but for screen-readers only?
* Should the print heading be removed from when users are printing?  Meaning, should we just remove this feature from when a page is printed, or is the goal to just remove this from the heading structure expressed to a screen-reader but retain it for a printed version of the page?

**Comment by jkeck** [comment]
Also, is there any desired state for what the screenreader nav should do WRT this particular component of the page and navigation?  Locally I decoupled the H1 from the dropdown component, but it didn't seem to have much of a discernible change to the screenreader's consumption of that section of the page (admittedly using my very limited screenreader experience).

**Comment by jvine** [comment]
> Should the content that's in the h1 now still remain...

Yes.

> Should the print heading be removed from when users are printing?

Yes. It was never necessary.

> Also, is there any desired state for what the screenreader nav should do 

I'm not certain. Camille & I are meeting with the SOAP online accessiblity consultant on Tuesday at 11 specifically to look at this section (you're welcome to join).

**Comment by jkeck** [comment]
Ah, yes, Camille did mention that. I wa going to try to come up w/ a potential proposal of what to do to address it.  But perhaps I'll just wait until after that meeting.

**Comment by jvine** [comment]
Some notes from meeting:
- remove `aria-haspopup`
- remove `role=menu`
- from `<h1>` from the dropdown control
- label the control to identify its purpose
- move a lower `<h1 class="sr-only">` to the top
- identify the SW logo link as the "home" link

Larger issues...
- make the labels of the search target dropdown and input clearer, and make them visible for voice-activated commands (Dragon)
- separate the "library resources" block into its own section ("aside"?)

And note that the unlabelled search input (from typeahead) did not announce itself to Jaws, even though the various audit tools report it.

**Comment by jkeck** [comment]
@jvine, when you say

> And note that the unlabelled search input (from typeahead) did not announce itself to Jaws, even though the various audit tools report it.

I wonder if you're meaning in apps like EarthWorks.  We don't implement the upstream typeahead so SearchWorks _shouldn't_ be experience that particular a11y issue (or at least the one that I was thinking of when we've talked about the unlabeled search inputs).

**Comment by jvine** [comment]
Ah. Never mind then.
